File: app/src/main/java/com/sxshi/retrofit/base/IBaseView.java
package com.sxshi.retrofit.base;

/**
 * 网络请求异常  统一处理相关
 */
public interface IBaseView {
    void showLoading(String message);

    void showLoading();

    void hideLoading();

    void onUnknownError(String error);

    void onTimeout();

    void onNetworkError();

    boolean isNetworkConnected();

    void onConnectionError();
}

File: app/src/main/java/com/sxshi/retrofit/base/IBaseView.kt
package com.sxshi.retrofit.base

/**
 * 网络请求异常  统一处理相关
 */
interface IBaseView {
    fun showLoading(message: String?)
    fun hideLoading()
    fun onUnknownError(message: String?)
    fun onTimeout()
    fun onNetworkError()
    fun onConnectionError()
}